File: ScaleformUI_Lua/src/Menus/PauseMenus/Elements/Columns/TextColumn.lua
TextColumn = {}
TextColumn.__index = TextColumn
setmetatable(TextColumn, { __index = PM_Column })
TextColumn.__call = function() return "TextColumn" end

function TextColumn.New(pos)
    local col = PM_Column.New(pos)
    return setmetatable(col, TextColumn)
end

-- function TextColumn:AddItem(item)
--     PM_Column.AddItem(self, item)
-- end

function TextColumn:SetDataSlot(index)
    if index > #self.Items then return end
    local item = self.Items[index]
    local label = "TXT_COL_LBL"
    BeginScaleformMovieMethod(ScaleformUI.Scaleforms._pauseMenu._pause.handle, "SET_DATA_SLOT")
    PushScaleformMovieFunctionParameterInt(0)
    PushScaleformMovieFunctionParameterInt(index - 1)
    PushScaleformMovieFunctionParameterInt(0)
    PushScaleformMovieFunctionParameterInt(index - 1)
    PushScaleformMovieFunctionParameterInt(0)
    PushScaleformMovieFunctionParameterInt(0)
    PushScaleformMovieFunctionParameterBool(true)
    AddTextEntry(label, item.label)
    BeginTextCommandScaleformString(label)
    EndTextCommandScaleformString_2()
    EndScaleformMovieMethod()
end

function TextColumn:UpdateSlot(index)
    if index > #self.Items then return end
    local item = self.Items[index]
    local label = "TXT_COL_LBL"
    BeginScaleformMovieMethod(ScaleformUI.Scaleforms._pauseMenu._pause.handle, "UPDATE_SLOT")
    PushScaleformMovieFunctionParameterInt(0)
    PushScaleformMovieFunctionParameterInt(index - 1)
    PushScaleformMovieFunctionParameterInt(0)
    PushScaleformMovieFunctionParameterInt(index - 1)
    PushScaleformMovieFunctionParameterInt(0)
    PushScaleformMovieFunctionParameterInt(0)
    PushScaleformMovieFunctionParameterBool(true)
    BeginTextCommandScaleformString(label)
    EndTextCommandScaleformString_2()
    EndScaleformMovieMethod()
end

File: ScaleformUI_Csharp/Menus/Pause Menus/Elements/Columns/TextColumn.cs
using ScaleformUI.Menus;
using ScaleformUI.PauseMenu;
using System.Security.Policy;

namespace ScaleformUI.PauseMenus.Elements
{
    public class TextColumn : PM_Column
    {
        public TextColumn(int position) : base(position)
        {
            VisibleItems = 16;
        }

        public override void SetDataSlot(int index)
        {
            if (index >= Items.Count)
                return;
            PauseMenuItem it = Items[index];
            var labels = it.Label.SplitLabel;
            BeginScaleformMovieMethod(Main.PauseMenu._pause.Handle, "SET_DATA_SLOT");
            PushScaleformMovieFunctionParameterInt(0);
            PushScaleformMovieFunctionParameterInt(index);
            PushScaleformMovieFunctionParameterInt(0);
            PushScaleformMovieFunctionParameterInt(index);
            PushScaleformMovieFunctionParameterInt(0);
            PushScaleformMovieFunctionParameterInt(0);
            PushScaleformMovieFunctionParameterBool(true);
			AddTextEntry("TXT_COL_LBL", it.Label.Label);
			BeginTextCommandScaleformString("TXT_COL_LBL");
			EndTextCommandScaleformString_2();
			EndScaleformMovieMethod();
        }

        public override void UpdateSlot(int index)
        {
            if (index >= Items.Count)
                return;
            PauseMenuItem it = Items[index];
            var labels = it.Label.SplitLabel;
            BeginScaleformMovieMethod(Main.PauseMenu._pause.Handle, "UPDATE_SLOT");
            PushScaleformMovieFunctionParameterInt(0);
            PushScaleformMovieFunctionParameterInt(index);
            PushScaleformMovieFunctionParameterInt(0);
            PushScaleformMovieFunctionParameterInt(index);
            PushScaleformMovieFunctionParameterInt(0);
            PushScaleformMovieFunctionParameterInt(0);
            PushScaleformMovieFunctionParameterBool(true);
			AddTextEntry("TXT_COL_LBL", it.Label.Label);
			BeginTextCommandScaleformString("TXT_COL_LBL");
			EndTextCommandScaleformString_2();
			EndScaleformMovieMethod();
        }
    }
}
